File: 실행/check_tables.py
import os
from supabase import create_client, Client
from dotenv import load_dotenv

load_dotenv(".env.local")

url = os.environ.get("NEXT_PUBLIC_SUPABASE_URL")
key = os.environ.get("NEXT_PUBLIC_SUPABASE_ANON_KEY")

if not url or not key:
    print("Error: Env vars not set")
    exit(1)

supabase: Client = create_client(url, key)
tables = ['schedules', 'breaks', 'holidays', 'events', 'partners']

print("--- Supabase Table Integrity Check ---")
for table in tables:
    try:
        response = supabase.table(table).select("*").limit(1).execute()
        print(f"✅ {table}: EXISTS")
    except Exception as e:
        if "404" in str(e) or "does not exist" in str(e).lower():
            print(f"❌ {table}: MISSING")
        else:
            print(f"⚠️ {table}: ERROR ({str(e)})")

File: 임시/check-tables.mjs
import { createClient } from '@supabase/supabase-js'
import dotenv from 'dotenv'
import path from 'path'

dotenv.config({ path: path.resolve(process.cwd(), '.env.local') })

const supabaseUrl = process.env.NEXT_PUBLIC_SUPABASE_URL
const supabaseAnonKey = process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY

const supabase = createClient(supabaseUrl, supabaseAnonKey)

async function checkTables() {
  const tables = ['schedules', 'breaks', 'holidays', 'events', 'partners']
  console.log('--- Table Status Check ---')
  for (const table of tables) {
    const { error } = await supabase.from(table).select('*').limit(1)
    if (error) {
      if (error.code === '42P01') {
        console.log(`❌ ${table}: Not Found`)
      } else {
        console.log(`⚠️ ${table}: Error (${error.message})`)
      }
    } else {
      console.log(`✅ ${table}: Exists`)
    }
  }
}

checkTables()
